es-app/src/CollectionSystemManager.cpp

```diff
@@ -225,127 +278,127 @@ void CollectionSystemManager::refreshCollectionSystems(FileData* file)
+{
+	this->system->getRootFolder()->sort(getSortTypeFromString(this->decl.defaultSort));
+	SystemData* systemViewToUpdate = CollectionSystemManager::get()->getSystemToView(this->system);
+	ViewController::get()->onFileChanged(systemViewToUpdate->getRootFolder(), FILE_SORTED);
+}
+
+
 // deletes all collection files from collection systems related to the source file
 void CollectionSystemManager::deleteCollectionFiles(FileData* file)
 {
 	// collection files use the full path as key, to avoid clashes
 	std::string key = file->getFullPath();
-	// find games in collection systems
-	std::map<std::string, CollectionSystemData> allCollections;
-	allCollections.insert(mAutoCollectionSystemsData.begin(), mAutoCollectionSystemsData.end());
-	allCollections.insert(mCustomCollectionSystemsData.begin(), mCustomCollectionSystemsData.end());
 
-	for(auto sysDataIt = allCollections.begin(); sysDataIt != allCollections.end(); sysDataIt++)
+	// find games in collection systems
+	for(auto & system : mCollectionSystems)
 	{
-		if (sysDataIt->second.isPopulated)
+		if (system.second->isPopulated)
 		{
-			const std::unordered_map<std::string, FileData*>& children = (sysDataIt->second.system)->getRootFolder()->getChildrenByFilename();
+			const std::unordered_map<std::string, FileData*>& children = (system.second->system)->getRootFolder()->getChildrenByFilename();
 
 			bool found = children.find(key) != children.end();
 			if (found) {
-				sysDataIt->second.needsSave = true;
+				system.second->needsSave = true;
 				FileData* collectionEntry = children.at(key);
-				ViewController::get()->getGameListView(sysDataIt->second.system).get()->remove(collectionEntry, false);
+				ViewController::get()->getGameListView(system.second->system).get()->remove(collectionEntry, false);
 			}
 		}
 	}
 }
 
-// returns whether the current theme is compatible with Automatic or Custom Collections
-bool CollectionSystemManager::isThemeGenericCollectionCompatible(bool genericCustomCollections)
-{
-	std::vector<std::string> cfgSys = getCollectionThemeFolders(genericCustomCollections);
-	for(auto sysIt = cfgSys.begin(); sysIt != cfgSys.end(); sysIt++)
-	{
-		if(!themeFolderExists(*sysIt))
-			return false;
-	}
-	return true;
-}
-
-bool CollectionSystemManager::isThemeCustomCollectionCompatible(std::vector<std::string> stringVector)
-{
-	if (isThemeGenericCollectionCompatible(true))
-		return true;
-
-	// get theme path
-	auto themeSets = ThemeData::getThemeSets();
-	auto set = themeSets.find(Settings::getInstance()->getString("ThemeSet"));
-	if(set != themeSets.end())
-	{
-		std::string defaultThemeFilePath = set->second.path.string() + "/theme.xml";
-		if (fs::exists(defaultThemeFilePath))
-		{
-			return true;
-		}
-	}
 
-	for(auto sysIt = stringVector.begin(); sysIt != stringVector.end(); sysIt++)
-	{
-		if(!themeFolderExists(*sysIt))
-			return false;
-	}
-	return true;
-}
+//TODO:  this method is only called by isThemeCustomCollectionCompatible(), which itself was never called?
```

Comment:
These were used originally to detect whether the theme was compatible and display a warning about it. If they're no longer used at all, feel free to delete.

Reply:
I will clean those up then!

Reply:
Thanks!